modular_skyrat/modules/xenos_skyrat_redo/code/xeno_types/sentinel.dm

```diff
@@ -0,0 +1,144 @@
+	var/button_base_icon = "neurospit"
+	/// The sound that should be played when the xeno actually spits
+	var/spit_sound = 'modular_skyrat/modules/xenos_skyrat_redo/sound/alien_spitacid.ogg'
+	shared_cooldown = MOB_SHARED_COOLDOWN_3
+	cooldown_time = 5 SECONDS
+
+/datum/action/cooldown/alien/acid/skyrat/IsAvailable()
+	return ..() && isturf(owner.loc)
+
+/datum/action/cooldown/alien/acid/skyrat/set_click_ability(mob/on_who)
+	. = ..()
+	if(!.)
+		return
+
+	to_chat(on_who, span_notice("You prepare your [projectile_name] gland. <B>Left-click to fire at a target!</B>"))
+
+	button_icon_state = "[button_base_icon]_1"
+	UpdateButtons()
+	on_who.update_icons()
+
+/datum/action/cooldown/alien/acid/skyrat/unset_click_ability(mob/on_who, refund_cooldown = TRUE)
+	. = ..()
+	if(!.)
+		return
+
+	if(refund_cooldown)
+		to_chat(on_who, span_notice("You empty your [projectile_name] gland."))
+
+	button_icon_state = "[button_base_icon]_0"
+	UpdateButtons()
+	on_who.update_icons()
+
+/datum/action/cooldown/alien/acid/skyrat/InterceptClickOn(mob/living/caller, params, atom/target)
+	. = ..()
+	if(!.)
+		unset_click_ability(caller, refund_cooldown = FALSE)
+		return FALSE
+
+	var/turf/user_turf = caller.loc
+	var/turf/target_turf = get_step(caller, target.dir)
+	if(!isturf(target_turf))
+		return FALSE
+
+	var/modifiers = params2list(params)
+	caller.visible_message(
+		span_danger("[caller] spits [projectile_name]!"),
+		span_alertalien("You spit [projectile_name]."),
+	)
+
+	if(acid_projectile)
+		var/obj/projectile/spit_projectile = new acid_projectile(caller.loc)
+		spit_projectile.preparePixelProjectile(target, caller, modifiers)
+		spit_projectile.firer = caller
+		spit_projectile.fire()
+		playsound(caller, spit_sound, 100, TRUE, 5, 0.9)
+		caller.newtonian_move(get_dir(target_turf, user_turf))
+		return TRUE
+
+	if(acid_casing)
+		var/obj/item/ammo_casing/casing = new acid_casing(caller.loc)
+		playsound(caller, spit_sound, 100, TRUE, 5, 0.9)
+		casing.fire_casing(target, caller, null, null, null, ran_zone(), 0, caller)
+		caller.newtonian_move(get_dir(target_turf, user_turf))
+		return TRUE
+
+	CRASH("Neither acid_projectile or acid_casing are set on [caller]'s spit attack!")
+
+/datum/action/cooldown/alien/acid/skyrat/Activate(atom/target)
+	return TRUE
+
+/obj/projectile/neurotoxin/skyrat
+	name = "neurotoxin spit"
+	icon_state = "neurotoxin"
+	damage = 40
+	paralyze = 0
+	damage_type = STAMINA
+	nodamage = FALSE
+	armor_flag = BIO
+
+/obj/projectile/neurotoxin/on_hit(atom/target, blocked = FALSE)
+	if(isalien(target))
+		nodamage = TRUE
+	return ..()
+
+/datum/action/cooldown/alien/acid/skyrat/lethal
+	name = "Spit Acid"
+	desc = "Spits neurotoxin at someone, burning them."
+	acid_projectile = /obj/projectile/neurotoxin/skyrat/acid
+	button_icon_state = "acidspit_0"
+	projectile_name = "acid"
+	button_base_icon = "acidspit"
+
+/obj/projectile/neurotoxin/skyrat/acid
+	name = "acid spit"
+	icon_state = "toxin"
+	damage = 25
```

Comment:
25 might be a bit much, but I haven't seen it in play yet, does it respect armor?

Reply:
as above, it is hard to spam abilities like these due to how much plasma it uses.